### DISCUSSION TITLE
How to define fonts used in book

### DISCUSSION THREAD

**Discussion by afduggirala** [johnfactotum/foliate#949] [General]
Hello,
I'm trying to read a book that contains code samples. The font used for the code samples looks tiny. I'm on Gnome 40.7 and on a high dpi screen that is being scaled at 300%. Is there a way for me to detect which style these fonts correspond to? Is there a way for me to make this font bigger? I know I can change the overall font size, but that makes regular (non-code) text look really big.
Thanks for your help and really nice software.
![Screenshot from 2022-02-13 19-57-19](https://user-images.githubusercontent.com/5265205/153783638-c85758c0-ea20-4f4d-af60-09166484b898.png)

**Version:**
 - Foliate version: 2.64
 - OS/Distribution and version: Fedora 34
 - Desktop environment: GNOME 40.7
 - Installation method: Fedora @system

**Comment by johnfactotum** [comment]
You can't. You have to edit the book's stylesheet.

There is a feature request for setting the monospace font: #222

**Comment by afduggirala** [comment]
I was able to change the stylesheet in Calibre. The change appears to have been effective, the monospace text appears bigger int the Calibre reader. However, in Foliate, the text size was not affected. Of course, I have made sure that I am looking at the newly edited book in Foliate, I think,
thank you,

**Comment by johnfactotum** [comment]
You can try using WebKit's inspector in Foliate to see which styles are applied to the element.

WeKit's Developer Tools can be accessed by going to the primary menu > _Advanced_ > _Enable Developer Tools_. After that, you can right click on any element and select _Inspect Element_.

**Comment by afduggirala** [comment]
Thank you John,
The inspector is showing me that the "pre" section style is still using a 0.8em font. I am pretty sure I edited this file to change this value to 1.0em. I will however edit the book again in Calibre to make sure this is the case. Is there any possibility that Foliate is keeping a copy of this book in some kind of cache? or its styes? (and that is why the changes are not taking effect in Foliate)
thanks again,

**Comment by johnfactotum** [comment]
>  Is there any possibility that Foliate is keeping a copy of this book in some kind of cache? or its styes? (and that is why the changes are not taking effect in Foliate)

There is a bug when merely reloading the book (#544). But it should work if you close Foliate then re-open the file.
